crates/pipeline-manager/src/integration_test.rs

```diff
@@ -2117,3 +2126,84 @@ async fn test_get_metrics() {
+    let response = config.get("/v0/pipelines/test/logs").await;
+    assert_eq!(response.status(), StatusCode::NOT_FOUND);
+
+    // Create pipeline
+    let request_body = json!({
+        "name": "test",
+        "description": "Description of the test pipeline",
+        "runtime_config": {},
+        "program_code": "CREATE TABLE t1(c1 INTEGER) with ('materialized' = 'true');",
+        "program_config": {}
+    });
+    let response = config.post("/v0/pipelines", &request_body).await;
+    assert_eq!(response.status(), StatusCode::CREATED);
+
+    // Retrieve logs (shutdown)
+    let mut response_logs = config.get("/v0/pipelines/test/logs").await;
+    assert_eq!(response_logs.status(), StatusCode::OK);
+    assert_eq!(
+        "LOGS STREAM END: no logs currently available (likely, the pipeline has not yet started)\n",
+        String::from_utf8(response_logs.body().await.unwrap().to_vec()).unwrap()
+    );
+
+    // Wait for its program compilation completion
+    config
+        .wait_for_compilation("test", 1, config.compilation_timeout)
+        .await;
+
+    // Start the pipeline in Paused state
+    let response = config.post_no_body("/v0/pipelines/test/pause").await;
+    assert_eq!(response.status(), StatusCode::ACCEPTED);
+
+    // Await it reaching paused status
+    config
+        .wait_for_deployment_status("test", PipelineStatus::Paused, config.start_timeout)
+        .await;
+
+    // Retrieve logs (paused)
+    let mut response_logs_paused = config.get("/v0/pipelines/test/logs").await;
+
+    // Start the pipeline
+    let response = config.post_no_body("/v0/pipelines/test/start").await;
+    assert_eq!(response.status(), StatusCode::ACCEPTED);
+    config
+        .wait_for_deployment_status("test", PipelineStatus::Running, config.resume_timeout)
+        .await;
+
+    // Retrieve logs (running)
+    let mut response_logs_running = config.get("/v0/pipelines/test/logs").await;
+
+    // Shut the pipeline down
+    let response = config.post_no_body("/v0/pipelines/test/shutdown").await;
+    assert_eq!(response.status(), StatusCode::ACCEPTED);
+    config
+        .wait_for_deployment_status("test", PipelineStatus::Shutdown, config.shutdown_timeout)
+        .await;
+
+    // Check the logs
+    assert_eq!(response_logs_paused.status(), StatusCode::OK);
+    assert_eq!(response_logs_running.status(), StatusCode::OK);
+    let logs1 = String::from_utf8(response_logs_paused.body().await.unwrap().to_vec()).unwrap();
+    let logs2 = String::from_utf8(response_logs_running.body().await.unwrap().to_vec()).unwrap();
+    assert_eq!(logs1, logs2);
+    assert!(logs1.ends_with("LOGS STREAM END: pipeline is being shutdown\n"));
+
+    // Retrieve logs (shutdown)
+    let mut response_logs = config.get("/v0/pipelines/test/logs").await;
+    assert_eq!(response_logs.status(), StatusCode::OK);
+    assert_eq!(
+        "LOGS STREAM END: no logs currently available (likely, the pipeline has not yet started)\n",
```

Comment:
```suggestion
        "LOG STREAM: no logs available (likely, the pipeline has not yet started)\n",
```

maybe this could just be empty instead of this string?

Reply:
I changed it to `LOG STREAM UNAVAILABLE: the pipeline has likely not yet started`
I think it is useful for the user of the logs endpoint to get back some feedback, rather than just empty, as the stream will not await till the pipeline starts running but already return.